File: frontend/src/types/asset.ts
export type AssetType = 'VIDEO' | 'IMAGE' | 'AUDIO' | 'TEMPLATE'

export interface Asset {
  id: number
  type: AssetType
  name: string
  fileUrl: string
  fileSize: number
  mimeType: string
  tags: string[]
  folderId: number | null
  thumbnail: string | null
  duration: number | null
  width: number | null
  height: number | null
  createdAt: string
}

export interface AssetFolder {
  id: number
  name: string
  parentId: number | null
  assetCount: number
  createdAt: string
}

export interface AssetFilter {
  type?: AssetType
  tags?: string[]
  folder?: number | null
  dateRange?: { startDate: string; endDate: string }
  search?: string
}

// Backend DTO types
export interface AssetResponse {
  id: number
  filename: string
  originalFilename: string | null
  fileUrl: string
  fileType: string
  fileSizeBytes: number | null
  mimeType: string | null
  tags: string[]
  folder: string
  width: number | null
  height: number | null
  durationSeconds: number | null
  createdAt: string | null
}

export interface AssetListResponse {
  assets: AssetResponse[]
  totalCount: number
}

export interface UpdateAssetRequest {
  tags?: string[]
  folder?: string
}

File: backend/onGo-domain/src/main/kotlin/com/ongo/domain/asset/Asset.kt
package com.ongo.domain.asset

import java.time.LocalDateTime

data class Asset(
    val id: Long? = null,
    val userId: Long,
    val filename: String,
    val originalFilename: String? = null,
    val fileUrl: String,
    /** 우리 버킷의 정확한 객체 키. 탈퇴 정리의 유일한 근거다. */
    val storageObjectKey: String? = null,
    val fileType: String,
    val fileSizeBytes: Long? = null,
    val mimeType: String? = null,
    val tags: List<String> = emptyList(),
    val folder: String = "default",
    val width: Int? = null,
    val height: Int? = null,
    val durationSeconds: Int? = null,
    val createdAt: LocalDateTime? = null,
)
